graphql_compiler/global_utils.py

```diff
@@ -1,8 +1,27 @@
 # Copyright 2017-present Kensho Technologies, LLC.
-from graphql import GraphQLList, GraphQLNamedType, GraphQLNonNull
+from dataclasses import dataclass
+from typing import Any, Dict
+
+from graphql import DocumentNode, GraphQLList, GraphQLNamedType, GraphQLNonNull
 import six
 
 
+@dataclass
+class QueryStringWithParameters:
+    """A query string and parameters that validate against the query."""
+
+    query_string: str
+    parameters: Dict[str, Any]
+
+
+@dataclass
+class ASTWithParameters:
+    """A query AST and parameters that validate against the query."""
+
+    query_ast: DocumentNode
+    parameters: Dict[str, Any]
```

Comment:
Moved from pagination module

Reply:
With https://github.com/kensho-technologies/graphql-compiler/pull/738 it might be a good idea to add something like `QueryWithPaginationAnalysis` which contains the query string, query ast, the cost estimate, the pagination capacity map, the metadata table, etc. This way we:
- avoid parsing to AST and printing again back to string
- make plumbing easier, reducing the number of parameters in most functions in this module